src/OpenTelemetry/Metrics/CounterMetricSdkBase.cs

```diff
@@ -59,7 +59,10 @@ internal BoundCounterMetric<T> Bind(LabelSet labelset, bool isShortLived)
             lock (this.bindUnbindLock)
             {
                 var recStatus = isShortLived ? RecordStatus.UpdatePending : RecordStatus.Bound;
-                boundInstrument = this.counterBoundInstruments.GetOrAdd(labelset, this.CreateMetric(recStatus));
+                if (!this.counterBoundInstruments.TryGetValue(labelset, out boundInstrument))
+                {
+                    boundInstrument = this.counterBoundInstruments.GetOrAdd(labelset, this.CreateMetric(recStatus));
```

Comment:
If the TryAdd fail, we will return the wrong boundInstrument.

Reply:
`GetOrAdd` doesn't guarantee anything different, does it? From the doc:
> Since a key/value can be inserted by another thread while valueFactory is generating a value, you cannot trust that just because valueFactory executed, its produced value will be inserted into the dictionary and returned. If you call GetOrAdd simultaneously on different threads, valueFactory may be called multiple times, but only one key/value pair will be added to the dictionary.

If we need a guarantee why not use a `Dictionary` in a lock?